backend/domain.py

```diff
@@ -0,0 +1,94 @@
+class Domain(object):
+    '''
+    The domain class represents the domain of the game, and is created by the user
+    in the domain json.
+
+    It includes the name of the domain, object types, predicate definitions,
+    and action definitions.
+    '''
+
+    def __init__(self):
+        """
+        Initializes a domain object with default values.
+        """
+        self.name = ""
+        self.object_types = []
+        self.predicates = []
+        self.actions = []
+
+    def are_valid_object_types(self, object_types):
```

Comment:
This should also return a boolean given the name of the function suggests so

Reply:
Changed accordingly